src/entities/repos/submission_repo.py

```diff
@@ -0,0 +1,77 @@
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncSession
+from typing import Any, List, TypeVar
+
+from entities.models import (
+    SubmissionDAO,
+    SubmissionDTO,
+    SubmissionState,
+    FilingPeriodDAO,
+    FilingPeriodDTO,
+    FilingDTO,
+    FilingDAO,
+)
+
+T = TypeVar("T")
+
+
+async def get_submissions(session: AsyncSession, filing_id: int = None) -> List[SubmissionDAO]:
+    async with session.begin():
+        stmt = select(SubmissionDAO)
+        if filing_id:
+            stmt = stmt.filter(SubmissionDAO.filing == filing_id)
+        results = await session.scalars(stmt)
+        return results.all()
+
+
+async def get_submission(session: AsyncSession, submission_id: int) -> SubmissionDAO:
+    return await query_helper(session, submission_id, SubmissionDAO)
+
+
+async def get_filing(session: AsyncSession, filing_id: int) -> FilingDAO:
+    return await query_helper(session, filing_id, FilingDAO)
+
+
+async def get_filing_period(session: AsyncSession, filing_period_id: int) -> FilingPeriodDAO:
+    return await query_helper(session, filing_period_id, FilingPeriodDAO)
+
+
+async def add_submission(session: AsyncSession, submission: SubmissionDTO) -> SubmissionDAO:
+    async with session.begin():
+        new_sub = SubmissionDAO(
+            filing=submission.filing,
+            submitter=submission.submitter,
+            state=SubmissionState.SUBMISSION_UPLOADED,
+            validation_ruleset_version="v1",
```

Comment:
this should probably be null at this point, the validation process should be the one that updates it.